streams/src/test/java/org/apache/kafka/test/KStreamTestDriver.java

```diff
@@ -109,10 +109,7 @@ public ProcessorContext context() {
 
     public void process(final String topicName, final Object key, final Object value) {
         final ProcessorNode prevNode = context.currentNode();
-        ProcessorNode currNode = topology.source(topicName);
-        if (currNode == null && globalTopology != null) {
-            currNode = globalTopology.source(topicName);
-        }
+        final ProcessorNode currNode = sourceNodeByTopicName(topicName);
```

Comment:
This is not introduced in this PR but I found the following comment can be improved as well. Maybe we can piggyback the fix in this PR as well ?

Basically, with the added lines in `send`, the `topicName` passed here should never be a changelog topic, otherwise there will be no source processor associated with it. So instead of the check on line 116, we should just check the returned `currNode` from line 112, and if it is null, skip the rest of the function directly.